### ISSUE TITLE
Sudden disconnect doesn't remove user

### ISSUE THREAD

**Issue by ThePhilgrim** [ThePhilgrim/MantaTail#45]
![Skärmavbild 2022-01-04 kl  23 18 34](https://user-images.githubusercontent.com/76887896/148131605-73407fe2-4aa1-4e16-9026-61f8bf577176.png)

After `PRIVMSG` was sent without a `:`, the client was disconnected. When trying to connect again with the same nick, it was already taken.

**Comment by Akuli** [comment]
The server should probably QUIT the user when an exception happens in `recv_loop()`.

**Comment by ThePhilgrim** [comment]
@Akuli This is not an exception that happens in `recv_loop()` though.

It's an `AssertionError` from `assert colon_privmsg.startswith(":")`.

Does this even need to be handled? Since this would highlight a bug in the code, and a client always sends with `:` in front of the message.

This error would therefore mean there is a bug in the client. Maybe send `error_not_enough_params()` ?

**Comment by Akuli** [comment]
There are a lot of cases like this that currently crash mantatail. I'll let you decide what you want to do with them. On the one hand, the server shouldn't error, no matter what, but on the other hand, it's usable for testing IRC clients even if it will error in some corner cases.

**Comment by ThePhilgrim** [comment]
@Akuli I have the feeling we can consider this issue resolved. I can't think of any cases Mantatail errors from disconnects anymore.
I am not sure I'm considering all cases though. What do you think?

**Comment by Akuli** [comment]
I agree that this can be closed. In #58, I ensured that `recv_loop()` always closes the connection properly, even if an error occurs (in a badly written handler function, for example). The code that handles actually disconnecting the user is in `send_queue_thread`, and it contains plenty of error handling.
